core/CMakeLists.txt

```diff
@@ -43,21 +43,37 @@ if (APPLE)
             src/main/c/osx/accept.c
     )
 elseif (UNIX)
-    include_directories($ENV{JAVA_HOME}/include/linux/)
-    message(${CMAKE_SYSTEM_PROCESSOR})
-    if (CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64")
-        set(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/binaries/armlinux)
-        set(AARCH64 1)
-    else ()
-        set(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/binaries/linux)
-    endif ()
-    set(
+    if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+        MESSAGE("Building for FreeBSD")
+        include_directories($ENV{JAVA_HOME}/include/freebsd/)
+        set(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/main/resources/binaries/freebsd)
+        set(
+            SOURCE_FILES ${SOURCE_FILES}
+            src/main/c/freebsd/kqueue.c
+            src/main/c/share/net.c
+            src/main/c/freebsd/affinity.c
+            src/main/c/freebsd/accept.c
+        )
+    else (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
```

Comment:
i guess i wasn't prepare to see else() with a condition inside? may be worth leaving this as `else()`